spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -269,7 +283,7 @@ public WriterCommitMessage commit() throws IOException {
         return new TaskCommit();
       }
 
-      DataFile dataFile = DataFiles.fromInputFile(file.toInputFile(), null, metrics);
+      DataFile dataFile = DataFiles.fromInputFile(file.toInputFile(), null, metrics, keyMetadata);
```

Comment:
I think it would make more sense to keep the encrypted file around as `this.file` and call `fromEncryptedOutputFile` here, rather than tracking the metadata and the file separately.